pkg/meta/meta.go

```diff
@@ -1599,7 +1617,50 @@ func (m *Meta) GetHistoryDDLJobsIterator(startJobID int64) (LastJobIterator, err
 
 // HLastJobIterator is the iterator for gets the latest history.
 type HLastJobIterator struct {
-	iter *structure.ReverseHashIterator
+	iter        *structure.ReverseHashIterator
+	schemaNames set.StringSet
+	tableNames  set.StringSet
+}
+
+var jobExtractFields = []string{"schema_name", "table_name"}
+
+// ExtractSchemaAndTableNameFromJob extract schema_name and table_name from encoded Job structure
+// Note, here we strongly rely on the order of fields in marshalled string, just like checkSubstringsInOrder
+// Exported for test
+func ExtractSchemaAndTableNameFromJob(data []byte) (schemaName, tableName string, err error) {
+	m, err := partialjson.ExtractTopLevelMembers(data, jobExtractFields)
+
+	schemaNameToken, ok := m["schema_name"]
+	if !ok || len(schemaNameToken) != 1 {
+		return "", "", errors.New("name field not found in JSON")
+	}
+	schemaName, ok = schemaNameToken[0].(string)
+	if !ok {
+		return "", "", errors.Errorf("unexpected name field in JSON, %v", schemaNameToken)
+	}
```

Comment:
in current usecase, we are always extract top level basic elements, we can abstract this part into a generic method